app/uk.gov.hmrc.helptosavestridefrontend/auth/StrideAuth.scala

```diff
@@ -67,4 +70,22 @@ trait StrideAuth extends AuthorisedFunctions with AuthRedirects { this: Frontend
       }
     }
 
+  def authorisedFromStrideWithDetails(action: Request[AnyContent] ⇒ OperatorDetails ⇒ Future[Result])(redirectCall: Call): Action[AnyContent] =
+    Action.async { implicit request ⇒
+      authorised(AuthProviders(PrivilegedApplication)).retrieve(allEnrolments and credentials and name and email) {
+        case enrolments ~ creds ~ name ~ email ⇒
+          val necessaryRoles: Option[List[Enrolment]] =
```

Comment:
there's a bit of duplication of code which can be factored out in here and in `authorisedFromStride`